akka-javasdk/src/main/java/akka/javasdk/agent/MemoryProvider.java

```diff
@@ -79,9 +82,11 @@ static LimitedWindowMemoryProvider limitedWindow() {
    *   <li>Use only last N messages from the history
    *   <li>Whether reading from memory is enabled
    *   <li>Whether writing to memory is enabled
+   *   <li>Applies memory filters {@link MemoryFilter}
    * </ul>
    */
-  record LimitedWindowMemoryProvider(Optional<Integer> readLastN, boolean read, boolean write)
+  record LimitedWindowMemoryProvider(
+      Optional<Integer> readLastN, boolean read, boolean write, List<MemoryFilter> filters)
```

Comment:
Now that this contain filters, it's a little strange that the name contains 'window' in it. 

It's looking more like a memory read/write strategy. 

I considered adding it to yet another memory provider, but it will also be about reading partial memory, limited by filter and by number of messages.

We let it sink for a while.